codespell_lib/tests/test_basic.py

```diff
@@ -393,6 +393,41 @@ def test_case_handling(tmpdir, capsys):
         assert f.read().decode('utf-8') == 'this has an ASCII error'
 
 
+def test_case_handling_in_fixes(tmpdir, capsys):
+    """Test that the case of fixes is similar to the mispelled word."""
+    d = str(tmpdir)
+    with open(op.join(d, 'dictionary_with_reason.txt'), 'w') as f:
+        fix = 'adopter, adaptor'
+        reason = 'these are different words!'
+        f.write(f'adoptor->{fix}, {reason}\n')
+    dictionary_name = f.name
+
+    # the mispelled word is entirely lowercase
+    with open(op.join(d, 'bad.txt'), 'w') as f:
+        f.write('early adoptor\n')
+    code, stdout, _ = cs.main('-D', dictionary_name, f.name, std=True)
+    # all suggested fixes must be lowercase too
+    assert fix in stdout
+
+    # the mispelled word is capitalized
+    with open(op.join(d, 'bad.txt'), 'w') as f:
+        f.write('Early Adoptor\n')
+    code, stdout, _ = cs.main('-D', dictionary_name, f.name, std=True)
+    # all suggested fixes must be capitalized too
+    assert ', '.join(w.strip().capitalize() for w in fix.split(',')) in stdout
```

Comment:
While this does make writing the test efficient, it means we're essentially just testing that `function(a) == function(a)`. Aside from you're using single rather than double quotes and the name of the fix variable. It doesn't clarify what exactly the output should look like etc.

Reply:
Indeed, the implementation of the function and the test "happen" to look alike. That was a bit on purpose, to make sure we are comparing the proper strings. I intended the test to detect future changes in functionality, not to thoroughly test the current code which is straightforward. But I get your point and agree that literal strings are more readable.

The whole file seems to be using simple quotes.

Reply:
The whole **test** file `test_basic.py` seems to be using simple quotes.

File `_codespell.py` mixes simple and double quotes. Is the logic that user-visible text gets double quotes and small literal strings for internal use get simple quotes? It's not clear to me.

Reply:
> The whole file seems to be using simple quotes.

Sorry this wasn't really a code comment, just a literal comment about the differences between the two.

> File `_codespell.py` mixes simple and double quotes. Is the logic that user-visible text gets double quotes and small literal strings for internal use get simple quotes? It's not clear to me.

I'm not sure, maybe. :shrug:

Reply:
Fixed this one, simple quotes everywhere in my patch – it's consistent with the above explanation and looks better indeed.